### PR TITLE
Proofread and update the readme

### LINKED ISSUE #232: Update readme according to the new release policy
Update readme according to https://www.tarantool.io/en/doc/latest/release/policy/

- [ ] Which basic images are used
- [ ] How to get Tarantool of the latest release, certain series or LTS version
- [ ] How to build on top of these images

### PR DESCRIPTION
**Pull request by NickVolynkin**
This PR has a number of changes:

* Info about new images based on Ubuntu 22.04.
* Recent changes in package naming.
* General proofreading and formatting.

Resolve #232

### CODE REVIEW

**Review by artembo — APPROVED** [review]
i'm okay, found one mistake, but please request a review from the doc team

**Comment by artembo** [comment]
Please add the PR description as well

File: README.md [hunk 1]
```diff
@@ -1,104 +1,159 @@
...
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [What is Tarantool](#what-is-tarantool)
+- [Quickstart](#quickstart)
+- [What's on board](#whats-on-board)
+  - [Included modules](#included-modules)
+  - [Data directories](#data-directories)
+  - [Convenience tools](#convenience-tools)
+- [Versions, tags, and release policy](#versions-tags-and-release-policy)
```
[the remark this answers is not in the document]
Comment (NickVolynkin) [reply README.md:12]: It's the Oxford comma, we use it in the docs. https://www.grammarly.com/blog/what-is-the-oxford-comma-and-why-do-people-care-so-much-about-it/

File: README.md [hunk 2]
```diff
@@ -1,104 +1,159 @@
...
+- [How to use these images](#how-to-use-these-images)
+  - [Start a Tarantool instance](#start-a-tarantool-instance)
+  - [Start a secure Tarantool instance](#start-a-secure-tarantool-instance)
+  - [Connect to a running Tarantool instance](#connect-to-a-running-tarantool-instance)
+  - [Start a master-master replica set](#start-a-master-master-replica-set)
+  - [Add application code with a volume mount](#add-application-code-with-a-volume-mount)
+  - [Build your own images](#build-your-own-images)
+- [Environment Variables](#environment-variables)
```
Comment (patiencedaur) [inline README.md:20]:
```suggestion
- [Environment variables](#environment-variables)
```

File: README.md [hunk 3]
```diff
@@ -1,104 +1,159 @@
...
+[documentation](https://www.tarantool.io/en/doc/latest/getting_started/getting_started_db/#using-a-docker-image).
 
+## What's on board
 
-# About this image
+The `tarantool/tarantool` images contain the Tarantool executable
+and a combination of [Lua modules](#included-modules) and utilities often used in production.
+They are designed to be a building block for modern services, and as such have a few
+design choices that set them apart from the systemd-controlled Tarantool.
+We check these modules for compatibility with the Tarantool version included in the image.
```
Comment (patiencedaur) [inline README.md:70]:
```suggestion
We check all these extensions for compatibility with the Tarantool version included in the image.
```
not to repeat "modules and utilities"

File: README.md [hunk 4]
```diff
@@ -1,104 +1,159 @@
...
+$ docker run --rm -t -i tarantool/tarantool:2.10.0
+$ docker run --rm -t -i tarantool/tarantool:2.10.0-centos7
+$ docker run --rm -t -i tarantool/tarantool:2.10.0-ubuntu
+```
 
-Entry-point script provided by this image uses environment variables
-to configure various "external" aspects of configuration, such as
-replication sources, memory limits, etc... If specified, they override
-settings provided in your code. This way you can use docker-compose or
+Entry-point script in each of these these images uses
```
Comment (patiencedaur) [inline README.md:81]:
```suggestion
The entrypoint script in each of these images uses
```
AFAIK, the Docker command is ENTRYPOINT

File: README.md [hunk 5]
```diff
@@ -1,104 +1,159 @@
...
+[environment variables](#environment-variables)
+to set various configuration options,
+such as replication sources, memory limits, and so on.
+If specified, the environment variables override the settings provided in your code.
+This way you can use `docker compose` or
 other orchestration and deployment tools to set those options.
```
Comment (patiencedaur) [inline README.md:88]:
```suggestion
This way, you can set options using `docker compose` or
other orchestration and deployment tools.

```

File: README.md [hunk 6]
```diff
@@ -1,104 +1,159 @@
...
-- [prometheus](https://github.com/tarantool/prometheus): Instrument code and export metrics to Prometheus monitoring
-- [mqtt](https://github.com/tarantool/mqtt): Client for MQTT message brokers
-- [gis](https://github.com/tarantool/gis): store and query geospatial data
-- [gperftools](https://github.com/tarantool/gperftools): collect CPU profile to find bottlenecks in your code
+- [avro-schema](https://github.com/tarantool/avro-schema): Apache Avro scheme for your data.
+- [connpool](https://github.com/tarantool/connpool): Keep a pool of connections to other Tarantool instances.
+- [curl](https://github.com/tarantool/curl): HTTP client based on libcurl.
+- [expirationd](https://github.com/tarantool/expirationd): Automatically delete tuples based on expiration time.
+- [gis](https://github.com/tarantool/gis): Store and query geospatial data.
+- [gperftools](https://github.com/tarantool/gperftools): Collect CPU profile to find bottlenecks in your code.
```
Comment (patiencedaur) [inline README.md:101]:
```suggestion
- [gperftools](https://github.com/tarantool/gperftools): Collect a CPU profile to find bottlenecks in your code.
```

File: README.md [hunk 7]
```diff
@@ -1,104 +1,159 @@
...
+- [http](https://github.com/tarantool/http): Embedded HTTP server with flask-style routing support.
```
Comment (patiencedaur) [inline README.md:102]:
```suggestion
- [http](https://github.com/tarantool/http): Embedded HTTP server with Flask-style routing support.
```

File: README.md [hunk 8]
```diff
@@ -1,104 +1,159 @@
...
+- [memcached](https://github.com/tarantool/memcached): Access Tarantool as if it was a Memcached instance.
```
Comment (patiencedaur) [inline README.md:103]:
```suggestion
- [memcached](https://github.com/tarantool/memcached): Access Tarantool as if it was a memcached instance.
```
product name

File: README.md [hunk 9]
```diff
@@ -1,104 +1,159 @@
...
+- [metrics](https://github.com/tarantool/metrics): Metric collection library for Tarantool.
+- [mqtt](https://github.com/tarantool/mqtt): Client for MQTT message brokers.
+- [mysql](https://github.com/tarantool/mysql): Query MySQL right from Tarantool.
+- [pg](https://github.com/tarantool/pg): Query PostgreSQL right from Tarantool.
+- [prometheus](https://github.com/tarantool/prometheus): Instrument code and export metrics to Prometheus monitoring.
+- [queue](https://github.com/tarantool/queue): Priority queues with TTL and confirmations.
+- [vshard](https://github.com/tarantool/vshard): Automatically distribute data across multiple instances.
```
Comment (patiencedaur) [inline README.md:110]: I'd propose a preferential change. I'd consider making the list items grammatically parallel, i.e. all noun phrases or all imperative sentences.
Example:
```
mysql: A tool to query MySQL right from Tarantool.
queue: Priority queues with TTL and confirmation.
```
or:
```
mysql: Query MySQL right from Tarantool.
queue: Create priority queues with TTL and confirmation.
```

File: README.md [hunk 10]
```diff
@@ -1,104 +1,159 @@
...
+If the module you need is not listed here, there is a good chance we can add it.
+Open an issue [on our GitHub](https://github.com/tarantool/docker).
 
-- `/var/lib/tarantool` is a volume containing operational data
-  (snapshots, xlogs and vinyl runs)
+### Data directories
 
-- `/opt/tarantool` is a place where users should put their lua
-  application code
+Mount a volume to these directories to make use of them.
```
Comment (patiencedaur) [inline README.md:118]:
```suggestion
Mount these directories as volumes:
```
You can mount a volume to a container but not to a directory

File: README.md [hunk 11]
```diff
@@ -1,104 +1,159 @@
...
+### Convenience tools
+
+- `console`: execute it without arguments to open an administrative
+  console to a running Tarantool instance.
+
+- `tarantool_is_up`: returns `0` if Tarantool has been initialized and
+  is operating normally.
 
 - `tarantool_set_config.lua`: allows you to dynamically change certain
   settings without the need to recreate containers.
```
Comment (patiencedaur) [inline README.md:134]:
```suggestion
- `console`: Execute without arguments to open an administrative
  console to a running Tarantool instance.

- `tarantool_is_up`: Returns `0` if Tarantool has been initialized and
  is operating normally.

- `tarantool_set_config.lua`: Allows you to dynamically change certain
  settings without the need to recreate containers.
```
Use consistent capitalization after colons in all sections. I corrected this snippet according to what I saw above.

File: README.md [hunk 12]
```diff
@@ -1,104 +1,159 @@
...
 
-# How to use this image
+## Versions, tags, and release policy
+
+The images are built and published on Docker Hub
+for each Tarantool release, and also for some beta
```
Comment (patiencedaur) [inline README.md:139]:
```suggestion
for each Tarantool release as well as for some beta
```

File: README.md [hunk 13]
```diff
@@ -175,50 +235,66 @@ networks:
 Start it like this:
 
 ``` console
-$ docker-compose up
+$ docker compose up
```
Comment (patiencedaur) [inline README.md:238]: Are you sure? Here's what my console says:
```
$ docker compose up
docker: 'compose' is not a docker command.
See 'docker --help'
```
I think it's `docker-compose up`

File: README.md [hunk 14]
```diff
@@ -175,50 +235,66 @@ networks:
...
 ```
 
-Where `/path/to/my/app` is a host directory containing lua code. Note
-that for your code to be actually run, you must execute the main script
-explicitly. Hence `tarantool /opt/tarantool/app.lua`, assuming that your
-app entry point is called `app.lua`.
+Where `/path/to/my/app` is a host directory containing lua code
+and `app.lua` is the entry point of your application.
+Note that for your code to actually run, you must execute the main script explicitly,
+which is done in the last line.
```
Comment (patiencedaur) [inline README.md:258]:
```suggestion
Here, `/path/to/my/app` is the host directory containing Lua code
and `app.lua` is the entry point of your application.
Note that for your code to run, you must execute the main script explicitly,
which is done in the last line.
```

File: README.md [hunk 15]
```diff
@@ -175,50 +235,66 @@ networks:
...
+
 Please pay attention to the format of `CMD`: unless it is specified in
 square brackets, the "wrapper" entry point that our Docker image
 provides will not be called. It will lead to inability to configure
 your instance using environment variables.
 
-## Environment Variables
+We recommend using an exact tag to build from,
+that is, `2.10.0` or `2.10.0-centos7`, but not just
+`2.10` or `latest`.
```
Comment (patiencedaur) [inline README.md:284]:
```suggestion
We recommend building from an image with a precise tag,
that is, `2.10.0` or `2.10.0-centos7`, not
`2.10` or `latest`.
```

File: README.md [hunk 16]
```diff
@@ -237,11 +313,13 @@ Tarantool. In the above example, it is set to "mysecretpassword".
...
 Optional. Specifying this variable will tell Tarantool to listen for
-incoming connections on a specific port. Default is 3301.
+incoming connections on a specific port. Default is `3301`.
 
 ### `TARANTOOL_PROMETHEUS_DEFAULT_METRICS_PORT`
 
-Optional. If specified tarantool will start http server on given port and expose prometheus `metrics` endpoint with common metrics (fibers, memory, network, replication, etc... )
+Optional. If specified tarantool will start http server on given port 
+and expose prometheus `metrics` endpoint with common metrics
+(fibers, memory, network, replication, etc.).
```
Comment (patiencedaur) [inline README.md:322]:
```suggestion
Optional. If specified, Tarantool will start an HTTP server on the provided port
and expose the Prometheus `metrics` endpoint with common metrics
(fibers, memory, network, replication, etc.).
```

File: README.md [hunk 17]
```diff
@@ -250,8 +328,8 @@ sources. Upon the start, Tarantool will attempt to connect to
 those instances, fetch the data snapshot and start replicating
 transaction logs. In other words, it will become a slave. For the
 multi-master configuration, other participating instances of
-Tarantool should be started with the same TARANTOOL_REPLICATION.
-(NB: applicable only to >=1.7)
+Tarantool should be started with the same `TARANTOOL_REPLICATION`.
+(NB: applicable only to versions later than 1.7)
```
Comment (patiencedaur) [inline README.md:332]:
```suggestion
(NB: Applicable only to versions later than 1.7.)
```

File: README.md [hunk 18]
```diff
@@ -260,61 +338,71 @@ Example:
...
-or UPDATE requests begin failing. Default is 268435456 (256
+actually store tuples, in bytes. When the limit is reached, `INSERT`
+or `UPDATE` requests begin failing. Default is `268435456` (256
 megabytes).
 
 ### `TARANTOOL_SLAB_ALLOC_FACTOR`
 
 Optional. Used as the multiplier for computing the sizes of memory
 chunks that tuples are stored in. A lower value may result in less
 wasted memory depending on the total amount of memory available
```
Comment (patiencedaur) [inline README.md:349]:
```suggestion
wasted memory, depending on the total amount of memory available
```

File: README.md [hunk 19]
```diff
@@ -260,61 +338,71 @@ Example:
...
 
 ### `TARANTOOL_MEMTX_MIN_TUPLE_SIZE`
 
 Optional. Size of the smallest allocation unit, in bytes. It can be
-decreased if most of the tuples are very small. Default is 16.
+decreased if most of the tuples are very small. Default is `16`.
 
 ### `TARANTOOL_CHECKPOINT_INTERVAL`
 
 Optional. Specifies how often snapshots will be made, in seconds.
```
Comment (patiencedaur) [inline README.md:365]:
```suggestion
Optional. Specifies how often snapshots are made, in seconds.
```

File: README.md [hunk 20]
```diff
@@ -260,61 +338,71 @@ Example:
...
+
+More details can be found in the
+[log module reference](https://www.tarantool.io/en/doc/latest/reference/configuration/#confval-log_format).
 
 ### `TARANTOOL_LOG_LEVEL`
 
-Optional. Default value is 5 (that means INFO). More details can be found in [logging log levels configuration](https://www.tarantool.io/en/doc/latest/reference/configuration/#cfg-logging-log-level)
+Optional. Default value is 5 (that means INFO).
+More details can be found in
+[logging log levels configuration](https://www.tarantool.io/en/doc/latest/reference/configuration/#cfg-logging-log-level).
```
Comment (patiencedaur) [inline README.md:389]:
```suggestion
[log levels configuration](https://www.tarantool.io/en/doc/latest/reference/configuration/#cfg-logging-log-level).
```
"Logging log" looks awkward

File: README.md [hunk 21]
```diff
@@ -260,61 +338,71 @@ Example:
...
+
+## Contributing
 
-# Reporting problems and getting help
+### Reporting problems and getting help
 
 You can report problems and request
 features [on our GitHub](https://github.com/tarantool/docker).
 
 Alternatively you may get help on our [Telegram channel](https://t.me/tarantool).
```
Comment (patiencedaur) [inline README.md:398]:
```suggestion
Alternatively, you may get help on our [Telegram channel](https://t.me/tarantool).
```

**Review by patiencedaur — COMMENTED** [review]
Left some language-related comments. Will push more proofread material as a commit, because GitHub doesn't allow commenting on unchanged lines.

File: README.md (hunk 13 above)
Comment (NickVolynkin) [reply README.md:238]: Nope, it's the new syntax. https://docs.docker.com/compose/#compose-v2-and-the-new-docker-compose-command

**Pull request merged**